### PR TITLE
The stickmen from the paper robe will now actually attack other players

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR removes the "neutral" faction from the stickmen made by the paper robe. This feature can be toggled if you just want your stickmen to mostly just vibe. It also slightly tweaks what you say when you summon a stickman with a paper robe.

The stickmen created by the paper robe have a maxHealth of 50 now (instead of 100). "Wild" stickmen and the stickmen summoned by Mjor the Creative aren't affected by this change.

The paper robe now has a cooldown between summons of 5 seconds (up from 3 seconds).

## Why It's Good For The Game

Basically, before this PR, stickmen would just kinda hang around and not do much, even if a traitor just shot their creator in the face. Now, they'll attack (on sight) most things that aren't their creator or a stickman that their creator has made (or something else that shares their creator's non-neutral faction(s)), so you can sic them on other players.

I basically just copied the code for this from dehydrated carp plushies, so any bugs with those will probably carry over to the paper robe's minions.

I halved the maxHealth of these things and increased the cooldown time of the robe because shitting out a 100 maxHealth minion every 3 seconds sounds like it'd be actually cancerous to fight against.

## Changelog
:cl: ATHATH
balance: The stickmen created by the paper robe will now attack players (but not their creator, of course) on their own. This feature can be toggled using the new "Toggle Stick Minion Aggressiveness" button. In other words, the paper robe isn't a glorified paperweight anymore.
balance: The stickmen created by the paper robe have a maxHealth of 50 now (instead of 100). "Wild" stickmen and the stickmen summoned by Mjor the Creative aren't affected by this change.
balance: The paper robe now has a cooldown between summons of 5 seconds (up from 3 seconds).
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Wait, holy shit, these things have 100 health, lemme halve that (for summoned ones only) real quick.

**Comment by ATH1909** [comment]
bam, done

**Comment by Shadowflame909** [comment]
arent these the weak ones without guns.

100 health makes sense doc. Especially since you gotta get shot like 30 times to get them.

**Comment by necromanceranne** [comment]
No it doesn't make sense. The robes have a cooldown of two seconds. You literally can produce an infinite quantity of bodies and no gun short of assault rifles are going to adequately get past the huge mob of dudes. I would highly recommend increasing the cooldown to a more reasonable amount if you do intend to make it actually attack players.

The health isn't the issue. It's the mob spam. Also I am fairly sure it spawns more than one stickman at a time. Been a while since I got the robes.

**Comment by Tlaltecuhtli** [comment]
cant you make it a toggle? the only time i saw them used was to cannon fodder xenos (they attack xenos )

**Comment by Tharcoonvagh** [comment]
Fixes #49134

**Comment by ExcessiveUseOfCobblestone** [comment]
this should be a toggleable similar to the hiero staff

**Comment by ATH1909** [comment]
> No it doesn't make sense. The robes have a cooldown of two seconds. You literally can produce an infinite quantity of bodies and no gun short of assault rifles are going to adequately get past the huge mob of dudes. I would highly recommend increasing the cooldown to a more reasonable amount if you do intend to make it actually attack players.
> 
> The health isn't the issue. It's the mob spam. Also I am fairly sure it spawns more than one stickman at a time. Been a while since I got the robes.

The cooldown is actually 3 seconds, and it spawns 1 stickman per "casting".

**Comment by ATH1909** [comment]
> Fixes #49134

Er, not quite. This doesn't affect "natural" stickmen or the stickmen from Mjor, ONLY the stickmen from the paper robe.

**Comment by ATH1909** [comment]
> this should be a toggleable similar to the hiero staff

That sounds like a good idea, I'll look into that.

**Comment by ATH1909** [comment]
Alright, I've added a "Toggle Stick Minion Aggressiveness" button, removed a sleep() from the code of the paper robe, and increased the paper robe's cooldown time between summons to 5 seconds (up from 3 seconds).

**Comment by Rohesie** [comment]
How about a max number of stickmen active at any time?

File: code/modules/clothing/suits/wiz_robe.dm [hunk 1]
```diff
@@ -146,38 +146,44 @@
...
 
-	usr.say("Rise, my creation! Off your page into this realm!", forced = "stickman summoning")
+	usr.say("Rise, my creation! Off of your page and into this realm!", forced = "stickman summoning")
 	playsound(src.loc, 'sound/magic/summon_magic.ogg', 50, TRUE, TRUE)
-	var/mob/living/M = new /mob/living/simple_animal/hostile/stickman(get_turf(usr))
-	var/list/factions = usr.faction
+	var/mob/living/M = new /mob/living/simple_animal/hostile/stickman/paperrobe(get_turf(usr)) //these guys have half the maximum health of a normal stickman, to make them less cancerous to fight against when spammed
+	var/list/factions = usr.faction.Copy()
+	if(angry_sticks)
+		for(var/F in factions)
```
Comment (optimumtact) [inline code/modules/clothing/suits/wiz_robe.dm:182]: you could just do `factions.remove("neutral")`

File: code/modules/clothing/suits/wiz_robe.dm [hunk 2]
```diff
@@ -146,38 +146,44 @@
...
+			if(F == "neutral")
+				factions -= F //this is so that they'll attack other players and NPCs and stuff, but still won't attack the person who made them or each other
 	M.faction = factions
-	src.robe_charge = FALSE
-	sleep(30)
-	src.robe_charge = TRUE
-	to_chat(usr, "<span class='notice'>\The robe hums, its internal magic supply restored.</span>")
+	timer = world.time + 50 //5 second cooldown between stickman spawns
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/suits/wiz_robe.dm:186]: time define

File: code/modules/clothing/suits/wiz_robe.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/clothing/suits/wiz_robe.dm]: thanks

**Comment by ATH1909** [comment]
Alright, I undid oranges's suggestion (because it causes the code to not compile) and went back to how dehydrated carp remove the neutral faction from the carp they spawn.

File: code/modules/clothing/suits/wiz_robe.dm [hunk 3]
```diff
@@ -146,38 +146,44 @@
...
+	usr.say("Rise, my creation! Off of your page and into this realm!", forced = "stickman summoning")
 	playsound(src.loc, 'sound/magic/summon_magic.ogg', 50, TRUE, TRUE)
-	var/mob/living/M = new /mob/living/simple_animal/hostile/stickman(get_turf(usr))
-	var/list/factions = usr.faction
+	var/mob/living/M = new /mob/living/simple_animal/hostile/stickman/paperrobe(get_turf(usr)) //these guys have half the maximum health of a normal stickman, to make them less cancerous to fight against when spam-summoned
+	var/list/factions = usr.faction.Copy()
+	if(angry_sticks)
+		for(var/F in factions)
+			if(F == "neutral")
+				factions -= F  //this is so that they'll attack other players and NPCs and stuff, but still won't attack the person who made them or each other
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/suits/wiz_robe.dm:184]: just do `factions -= neutral`, if it doesnt exist it shouldnt freak out.

**Comment by Qustinnus** [comment]
if you're going to change the balance on a mob it should be in the title.

File: code/modules/clothing/suits/wiz_robe.dm (hunk 3 above)
Comment (ATH1909) [reply code/modules/clothing/suits/wiz_robe.dm:184]: are you absolutely, 100% sure about this?

**Comment by ATH1909** [comment]
> if you're going to change the balance on a mob it should be in the title.

it's in the changelog

and again, it ONLY APPLIES TO THE STICKMEN FROM THE ROBE. THE ONES IN THE SPACE RUIN (and the ones summoned by Mjor) ARE NOT AFFECTED.

File: code/modules/clothing/suits/wiz_robe.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/clothing/suits/wiz_robe.dm:184]: it is just a list.

Comment (Rohesie) [reply code/modules/clothing/suits/wiz_robe.dm:184]: `LAZYREMOVE(factions, "neutral")` should work alright
If the element is not on the list nothing will happen.

**Comment by Rohesie** [comment]
Are you still working on this?
Have you addressed the balance concern of infinite stickmen?

**Comment by kriskog** [comment]
merge conflict

**Comment by Rohesie** [comment]
Are you still planning on working on this?

**Comment by Rohesie** [comment]
Still no max cap of active stick figures at any given time?

**Comment by ExcessiveUseOfCobblestone** [comment]
You have generated new atom-bound variables/procs without proper documentation.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

File: code/modules/clothing/suits/wiz_robe.dm [hunk 4]
```diff
@@ -1,231 +1,235 @@
...
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
+	resistance_flags = FLAMMABLE
+
+/obj/item/clothing/suit/wizrobe/paper //fun fact: because this is a subtype of normal wizard robes, a normal wizard can actually cast spells in this
+	name = "papier-mache robe" // no non-latin characters!
+	desc = "A robe held together by various bits of clear-tape and paste."
+	icon_state = "wizard-paper"
+	inhand_icon_state = "wizard-paper"
+	var/timer = 0 //when we'll next be able to use the summon stickman ability
+	var/angry_sticks = TRUE // have you set the robe to make stickmen who WON'T have the neutral faction? note that this WON'T affect stickmen that (whom?) you've already made
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/suits/wiz_robe.dm:155]: new variables should be below inherited

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Cobby's review + docs

**Comment by Rohesie** [comment]
If you re-add it, please add a cap to how many of these can be alive at the same time.

**Pull request closed without merging**